snap/snapcraft.yaml

```diff
@@ -0,0 +1,55 @@
+name: k3s
+summary: Lightweight Kubernetes.
+description:|  k3s is wrapped in a simple package that reduces the dependencies and steps needed to run a production Kubernetes cluster. 
+  
+adopt-info: k3s
+
+grade: stable
+confinement: classic
+
+architectures:
+  - build-on: amd64
+  - build-on: armhf
+  - build-on: arm64
+
+apps:
+  k3s:
+    command: bin/k3s
+    plugs:
+      - home
+      - network
+      - removable-media
+
+parts:
+  k3s:
+plugin: nil 
+    source: https://github.com/rancher/k3s.git
+    source-type: git
+    override-pull: |
+      git clone https://github.com/rancher/k3s.git src/github.com/rancher/k3s
+       cd src/github.com/rancher/k3s
+      last_committed_tag="$(git describe --tags --abbrev=0)"
+      last_committed_tag_ver="$(echo ${last_committed_tag} | sed 's/v//')"
+      last_released_tag="$(snap info $SNAPCRAFT_PROJECT_NAME | awk '$1 == "beta:" { print $2 }')"
+      # If the latest tag from the upstream project has not been released to
+      # beta, build that tag instead of master.
+      if [ "${last_committed_tag_ver}" != "${last_released_tag}" ]; then
+        git fetch
+        git checkout "${last_committed_tag}"
+      fi
+      snapcraftctl set-version "$(git describe --tags | sed 's/v//')"
+      override-build: |
+      export GOPATH=$PWD
+      cd src/github.com/rancher/k3s
+      env CGO_ENABLED=0 GOOS=linux \
+      go build --ldflags "-s -w \
+        -X 'github.com/rancher/k3s/version.GitCommit=$(git rev-list -1 HEAD)' \
+        -X 'github.com/rancher/k3s/version.Version=$(git describe --tags --abbrev=0)'" \
+        -a -installsuffix cgo -o $SNAPCRAFT_PART_INSTALL/bin/k3s
```

Comment:
I think this is missing containerd.  Does this fully work?

Reply:
I need to update it